src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,38 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property the field to get value for
+     * @return mixed
+     */
+    public function getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
+            throw new \BadMethodCallException('Property "' . $property . '" not found.');
+        }
+
+        return $value;
+    }
+
+    /**
+     * Check if property exists in entity even if it is null
+     *
+     * @param string $property the field to check for existence
+     * @return bool
+     */
+    public function propertyExists($property)
+    {
+        $hasProperty = ($this->has($property) || $this->isDirty($property));
+        if (!$hasProperty && array_key_exists($property, $this->_properties)) {
```

Comment:
What about virtual properties? `method_exists()`?

Reply:
I did that. If the virtual property returns "null" it is considered as not existing. I think that covers all cases. I just thought that a virtual property which returns null makes no sense.

Reply:
I think it depends on how you use it and propertyExists does not tell you if it checks for virtual properties or not.

`public function propertyExists($property, $alsoCheckForVirtual = true)`
... would and depending on the core teams preferences the default could also be false.

Reply:
You are right. Feel free to update it ;-)